examples/cis3-nft-sponsored-txs/src/lib.rs

```diff
@@ -382,6 +360,16 @@ type ContractError = Cis2Error<CustomContractError>;
 
 type ContractResult<A> = Result<A, ContractError>;
 
+/// Mapping CustomContractError to ContractError
+impl From<CheckAccountSignatureError> for CustomContractError {
```

Comment:
The reason that you have to unwrap below is that your contract error type is `ContractError`, which is 
```
Cis2Error<CustomContractError>;
```
So you want a 
```
From<CheckAccountSignatureError> for Cis2Error<CustomContractError>
```

as well.

I think we should add this to concordium-std, like we have these from instances 
```
impl<X> From<NewReceiveNameError> for Cis2Error<X>
where
    X: From<NewReceiveNameError>,
{
    #[inline]
    /// Converts the error by wrapping it in [Self::Custom].
    fn from(err: NewReceiveNameError) -> Self { Cis2Error::Custom(X::from(err)) }
}
```
already.

Reply:
ok, It should be added to `concordium-cis2` library instead of `concordium-std`. I added it.